File: leetcode_cpp/59_螺旋矩阵II.cpp
//
// Created by 16517 on 2024/7/11.
//
#include <vector>
using namespace std;


class Solution {
public:
    vector<vector<int>> generateMatrix(int n) {
        vector<vector<int>> matrix(n,vector<int>(n));
        int offset = 0;
        int i = 0, j = 0;
        int count = 1;
        while (offset < (n / 2)) {
            for (j = offset; j < n - 1 - offset; j++) {
                matrix[offset][j] = count++;
            }
            for (i = offset; i < n - 1 - offset; i++) {
                matrix[i][j] = count++;
            }
            for (; j > offset; j--) {
                matrix[i][j] = count++;
            }
            for (; i > offset; i--) {
                matrix[i][j] = count++;
            }
            offset++;
        }
        if (n % 2 != 0) {
            matrix[n/2][n/2] = count;
        }
        return matrix;
    }
};

File: leetcode_rust/src/59_螺旋矩阵II.rs
impl Solution {
    pub fn generate_matrix(n: i32) -> Vec<Vec<i32>> {
        let mut matrix = vec![vec![0; n]; n];
        let mut offset = 0;
        let mut count = 1;
        let mut i = 0;
        let mut j = 0;
        while offset < n / 2 {
            for j in offset..n - 1 - offset {
                matrix[offset][j] = count;
                count += 1;
            }
            for i in offset..n - 1 - offset {
                matrix[i][j] = count;
                count += 1;
            }
        }
    }
}